File: backend/Models/CustomerReview.cs
using System.ComponentModel.DataAnnotations;

namespace Travyle.Api.Models;

public class CustomerReview
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid UserId { get; set; }
    public User? User { get; set; }

    [Required]
    public Guid TourId { get; set; }

    [Range(1, 5)]
    public int Rating { get; set; } = 5;

    [Required]
    [MaxLength(1000)]
    public string Comment { get; set; } = string.Empty;

    public bool IsVerified { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: mobile/lib/models/customer_review.dart
class CustomerReviewModel {
  final String id;
  final String userId;
  final String? userName;
  final String tourId;
  final int rating;
  final String comment;
  final bool isVerified;
  final DateTime createdAt;

  CustomerReviewModel({
    required this.id,
    required this.userId,
    this.userName,
    required this.tourId,
    required this.rating,
    required this.comment,
    required this.isVerified,
    required this.createdAt,
  });

  factory CustomerReviewModel.fromJson(Map<String, dynamic> json) {
    return CustomerReviewModel(
      id: json['id'] ?? '',
      userId: json['userId'] ?? '',
      userName: json['userName'],
      tourId: json['tourId'] ?? '',
      rating: json['rating'] ?? 5,
      comment: json['comment'] ?? '',
      isVerified: json['isVerified'] ?? true,
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}
